### PR TITLE
Update 15-3Sum.js

### PR DESCRIPTION
**Pull request by brentonjackson**
Refactor solution to be more readable and reflects JavaScript implementation of two pointer method in video

[//]: # "Pull Request Template"
[//]: # "Replace the placeholder values in the template below"

- **File(s) Modified**: _15-3Sum.js_
- **Language(s) Used**: _javascript_
- **Submission URL**: _https://leetcode.com/submissions/detail/836200467/_

[//]: # "Getting the Submission URL"
[//]: # "Go to the leetcode [`Submissions tab`](https://user-images.githubusercontent.com/71089234/180188604-b1ecaf90-bf27-4fd6-a559-5567aebf8930.png)"
[//]: # "and [click on the `Accepted` status of your submission.](https://user-images.githubusercontent.com/71089234/180189321-1a48c33f-aa65-4b29-8aaa-685f4f5f8c9e.png)]"
[//]: # "Finally copy the URL from the nav bar, it should look like https://leetcode.com/submissions/detail/xxxxxxxxx/"

### CODE REVIEW

File: javascript/15-3Sum.js [hunk 1]
```diff
@@ -4,49 +4,30 @@
...
-        if (isTarget) {
-            sums.push([-target, leftVal, rightVal]);
-            left++;
-            right--;
-
-            while (left < right && isPrevDuplicate(nums, left)) left++;
-            while (left < right && isNextDuplicate(nums, right)) right--;
-
-            continue;
+var threeSum = function(nums) {
```
Comment (aakhtar3) [inline javascript/15-3Sum.js:7]: You should create a new function below so that both solutions are present

Comment (brentonjackson) [reply javascript/15-3Sum.js:7]: will do

File: javascript/15-3Sum.js [hunk 2]
```diff
@@ -1,3 +1,37 @@
...
+ * @return {number[][]}
+ */
+var threeSum = function(nums) {
+    nums.sort((a, b) => a - b );
+    let outputArr = [];
+    let numOne, numTwo, numThree;
+    for (let i = 0; i < nums.length; i++) {
+        if (i > 0 && nums[i] == nums[i-1]) continue;
+        numOne = nums[i];
+        // do twoSum
```
Comment (aakhtar3) [inline javascript/15-3Sum.js:14]: Suggestion: Create a function called twoSum() and delete your comment

File: javascript/15-3Sum.js [hunk 3]
```diff
@@ -1,3 +1,37 @@
...
+        let [left, right] = [i+1, nums.length - 1];
+        while (left < right) {
+            // find two sums that equal 0 when added to first number
+            [numTwo, numThree] = [nums[left], nums[right]];
+            // continue if duplicate
```
Comment (aakhtar3) [inline javascript/15-3Sum.js:19]: Suggestion: Create Boolean isDuplicate and delete comment
```js
const isDuplicate = ...
if (isDuplicate) { left; continue; }
```

File: javascript/15-3Sum.js [hunk 4]
```diff
@@ -1,3 +1,37 @@
...
+ * Time O(N ^ 2) | Space O(N)
+ * @param {number[]} nums
+ * @return {number[][]}
+ */
+var threeSum = function(nums) {
+    nums.sort((a, b) => a - b );
+    let outputArr = [];
+    let numOne, numTwo, numThree;
+    for (let i = 0; i < nums.length; i++) {
+        if (i > 0 && nums[i] == nums[i-1]) continue;
```
Comment (aakhtar3) [inline javascript/15-3Sum.js:12]: Suggestion: Create function since this logic is referenced twice.
```js
const isDuplicate = (nums, index) => ...
```

**Pull request closed without merging**